### PR TITLE
Basic Web Structure

### PR DESCRIPTION
**Pull request by mohamedelhefni**
## Changes
* Introduce a shared network with the bridge driver.
* Create a shared volume for the backend service.
* Implement nodemon for enabling automatic hot reloading.
* Add Basic Web Structure and add Chakra UI to 
* Add React dom router with basic Layout

### CODE REVIEW

**Comment by xUser5000** [comment]
Awesome work! You did a really good job.

I just added my two cents regarding docker and npm (see the commits I added). I think this is robust enough: other devs could just `docker-compose up` or `docker-compose build` (whenever we add some packages) without having to know any details about docker.

I'll ping Hamdy to look at the frontend structure. In the meantime, why don't you dockerize the frontend as well and add it to the docker-compose workflow?

Again, thanks a lot

**Comment by xUser5000** [comment]
Also, if you have time, could you create another PR that contains a good README to include the name of the project, logo, how to install and run, what dependencies are needed (I assume docker should be one of them), and relevant links on how to install them. All of this is for the sake of making it easier to onboard other devs.

Thanks

File: docker-compose.yml [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
+    networks:
+      - my_network
 
   db:
     image: postgres:16-alpine
     environment:
       - POSTGRES_USER=user
       - POSTGRES_PASSWORD=passowrd
-      - POSTGRES_DB=database
+      - POSTGRES_DB=databaseo
```
Comment (mohamedelhefni) [inline docker-compose.yml:23]: typo

Comment (xUser5000) [reply docker-compose.yml:23]: Feel free to fix, but we will come to that later

**Review by hamdymohamed162002 — APPROVED** [review]
good

**Pull request merged**